src/main/java/org/javarosa/xpath/XPathConditional.java

```diff
@@ -51,6 +51,11 @@ public Object evalRaw(DataInstance model, EvaluationContext evalContext) {
             if (xpath != null) {
                 throw new XPathUnsupportedException(xpath);
             } else {
+                if (evalContext.isConstraint) {
+                    String contextMessage = String.format(" Question with id: \'%s\' has issue with condition expression: \'%s\'",
+                            evalContext.getContextRef().getNameLast(), expr);

```

Comment:
@ctsims Do you know if `evalContext.getContextRef().getNameLast()` will always give us the question id here ? If not, I think we should skip showing question id in the message.